compiler/shared_ast/expr.ml

```diff
@@ -253,6 +253,9 @@ let typed = Typed { pos = Pos.no_pos; ty = TLit TUnit, Pos.no_pos }
 (* - Predefined types (option) - *)
 
 let option_enum = EnumName.fresh [] ("Eoption", Pos.no_pos)
+
+(* Warning: order of these definitions is important, binary injection assumes
```

Comment:
What's binary injection? The way they're stored inside an `EnumConstructor.Map.t` ?

Reply:
Ah I mean the way we make compiled modules and the interpreter interact by translating between runtime OCaml values and AST value nodes on the fly. I added code to translate options, but it requires this assumption.